generate_kubernetes_enums.js

```diff
@@ -0,0 +1,145 @@
+        let l = lines[i];
+        const m  = /type\s+(\w+)\s+string/g.exec(l.trim());
+        if (m) {
+            enumTypes.push(m[1]);
+            continue;
+        }
+        const type =  /type\s+(\w+)\s+struct\s{/g.exec(l);
+        if (type) {
+            i++;
+            typeList[type[1]] = [];
+            while (lines[i] !== '}') {
+                if (lines[i].includes('}')) {
+                    throw new Error('unexpected quote' + lines[i]);
+                }
+                const field = parse_protocol_buf(lines[i]);
+                if ((!field || !field.name) && lines[i] !== 'metav1.TypeMeta `json:",inline"`') {
+                    throw new Error(`invalid line ${lines[i]}`)
+                } else {
+                    if (enumTypes.indexOf(field.type) >= 0) {
+                       typeList[type[1]].push(field);
+                    }
+                }
+
+                i++;
+            }
+            continue;
+        }
+        if (/const\s+\(/g.exec(l)) {
+            i++;
+            while (lines[i] !== ')') {
+                if (lines[i].includes(')')) {
+                    throw new Error('unexpected quote' + lines[i]);
+                }
+                const n = /(\w+)\s+(\w+)\s+=\s+"(\w+)"/.exec(lines[i]);
+                if (n) {
+                    const typeName = n[2];
+                    if (typeName !== 'string') {
+                        if (enumTypes.indexOf(typeName)<0)  {
+                            throw new Error(`invalid type ${typeName}`);
+                        }
+                        if (!enumTypeList[typeName]) {
+                            enumTypeList[typeName] = [n[3]];
+                        } else {
+                            enumTypeList[typeName].push(n[3]);
+                        }
+                    }
+
+                } else {
+                    // ignore
+                }
+                i++;
+            }
+        }
+    }
+    const json = JSON.parse(swagger_json);
+    const map = {};
+
+    _.each(Object.keys(typeList), key => {
+        let t = typeList[key];
+        _.each(t, field => {
+            if (enumTypeList[field.type]) {
+                // console.log('Got:', key, field.name, enumTypeList[field.type].join('|'))
+                map[key + '!' + field.name] = {
+                    enums: enumTypeList[field.type]
+                }
+            }
+        });
+    });
+    let current;
+    const enums = {};
+    _.each(json.definitions, (definitions, key) => {
+        current = key.split('.').slice(-1)[0];
+        enums[key] = {};
+        _.each(definitions.properties, (propValue, propKey) => {
+            if (map[current + '!' + propKey]) {
+                if (!propValue.enum) {
+                    propValue.enum = map[current + '!' + propKey].enums;
+                    enums[key][propKey] = propValue.enum;
+                }
+            }
+        });
+        if (Object.keys(enums[key]).length === 0) {
```

Comment:
I feel this would be clearer if you built the object in the `each` block as a named variable (extracting to a function and returning the built object), then assigned it to `enums[key]` if the object was non-empty.